### PR TITLE
Move code coverage to own github workflow

### PR DESCRIPTION
**Pull request by anjastrunk**
Move code coverage to  separate GitHub workflow to add code coverage overview to PR, always.

### CODE REVIEW

File: coveragerc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[run]
+source = creator
+
+[report]
+show_missing = True
+skip_empty = True
+
+[xml]
+output = coverage.xml
```
Comment (martinmo) [inline coveragerc:9]: This file should be merged with the hidden `.coveragerc` file. I think the non-hidden file isn't read by coverage (for example, the `show_missing` switch doesn't seem to be active when I run `coverage report` without the `-m` switch locally).

File: .github/workflows/build-python-app.yml [hunk 2]
```diff
@@ -35,23 +35,27 @@ jobs:
...
-          hide_complexity: true
+          hide_complexity: false
           indicators: true
           output: both
           thresholds: '90 95'
 
+      - name: add threshold to coverage.md
+        run: |
+          cat code-coverage-results.md
+
```
Comment (martinmo) [inline .github/workflows/build-python-app.yml:58]: The job name and actions do not align: `cat …` doesn't add a threshold, it only prints the markdown file to stdout.

**Review by martinmo — CHANGES_REQUESTED** [review]
I propose to stick with one workflow, `build-python-app.yml`, to avoid duplication:

If I understand it correctly, you added the new workflow because you want to have the PR comment to be created even if the coverage is below the threshold?

If this is this case, you can set [`jobs.<job-id>.strategy.fail-fast`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) to `false` like this:

```yaml
jobs:
  build-python-app:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
    steps:
    # ...
```

(The default is `true` which is the reason the steps after `CodeCoverageSummary` aren't executed it the threshold isn't met.)

Last, I propose you add `.gx-credentials/` to `.gitignore` so the credentials will not be committed unintentionally.

File: coveragerc (hunk 1 above)
Comment (anjastrunk) [reply coveragerc:9]: Yes, you are right. I merged `coveragerc` in `.coveragerc`

File: .github/workflows/build-python-app.yml (hunk 2 above)
Comment (anjastrunk) [reply .github/workflows/build-python-app.yml:60]: Removed job step `add threshold to coverage.md`

**Comment by martinmo** [comment]
> If this is this case, you can set [`jobs.<job-id>.strategy.fail-fast`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) to `false` like this:
> 
> ```yaml
> jobs:
>   build-python-app:
>     runs-on: ubuntu-latest
>     strategy:
>       fail-fast: false
>     steps:
>     # ...
> ```

@anjastrunk in addition to disabling `fail-fast`, one needs to add `!cancelled()` to the `if` clause of the steps following the code coverage step, see [bca4ffd](https://github.com/SovereignCloudStack/gx-credential-generator/pull/91/commits/bca4ffd509e3ecdb1ac9ad45e3bf3e157b2a7dc3) of the test PR https://github.com/SovereignCloudStack/gx-credential-generator/pull/91. Sorry for the incomplete initial suggestion!

File: .github/workflows/build-python-app.yml [hunk 3]
```diff
@@ -17,8 +17,12 @@ permissions:
   pull-requests: write
 
 jobs:
-  build:
+  build-python-app:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        experimental: [false]
```
Comment (martinmo) [inline .github/workflows/build-python-app.yml:25]:
```suggestion
```

**Comment by anjastrunk** [comment]
> > If this is this case, you can set [`jobs.<job-id>.strategy.fail-fast`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) to `false` like this:
> > ```yaml
> > jobs:
> >   build-python-app:
> >     runs-on: ubuntu-latest
> >     strategy:
> >       fail-fast: false
> >     steps:
> >     # ...
> > ```
> 
> @anjastrunk in addition to disabling `fail-fast`, one needs to add `!cancelled()` to the `if` clause of the steps following the code coverage step, see [bca4ffd](https://github.com/SovereignCloudStack/gx-credential-generator/pull/91/commits/bca4ffd509e3ecdb1ac9ad45e3bf3e157b2a7dc3) of the test PR #91. Sorry for the incomplete initial suggestion!

@martinmo Many thanks for your support. Now it works as expected.

**Review by martinmo — APPROVED** [review]
The changes are good to go in my opinion! Only the title of the PR should be changed to reflect what was actually done.

**Pull request merged**